uncorrelate/statistical_inefficiency_dhdl.py

```diff
@@ -50,13 +54,22 @@ def uncorrelate(self, dfs, lower):
             dl.append(dli)
 
         uncorrelated_dfs = []
-        for dhdl_, l, df in zip(self.dhdls, dl, dfs):
+        print("Number of correlated and uncorrelated samples (Method=%s):\n\n%6s %12s %12s %12s\n" % ("dHdl", "State", "N", "N_k", "N/N_k"))
+        for idx, (dhdl_, l, df) in enumerate(zip(self.dhdls, dl, dfs)):
             ind = np.array(l, dtype=bool)
             ind = np.array(ind, dtype=int)
             dhdl_sum = dhdl_.dot(ind)
-            uncorrelated_dfs.append(alchemlyb.preprocessing.statistical_inefficiency(df, dhdl_sum, lower, conservative=False))
+            uncorrelated_df = alchemlyb.preprocessing.statistical_inefficiency(df, dhdl_sum, lower, conservative=False)
+            N, N_k = len(df), len(uncorrelated_df)
+            g = N/N_k
+            print("%6s %12s %12s %12.2f" % (idx, N, N_k, g))
+            if N_k < self.uncorr_threshold:
+                print("WARNING: Only %d uncorrelated samples found at lambda number %d; proceeding with analysis using correlated samples..." % (N_k, idx))
```

Comment:
It might be better to use the warning instead of print the warning.

```
import warning
warnings.warn("Only %d uncorrelated samples found at lambda number %d")
```